### ISSUE TITLE
Aux Channel Unit Schema: Tier1 and Tier2 Types missing in lots of 2014-2016 Visits 

### ISSUE THREAD

**Issue by volkcj** [SouthForkResearch/CHaMP_Metrics#99]
This is a bit of a hot mess--here's a little more info:

- 2014 has some Tier 1 and Tier 2 types: e.g. VisitID 2538 has Tier1 and Tier 2 types, and Visit ID 2334 doesn't.  
- 2015 and 2016 Tier 1 and Tier 2 types are all null.

- There are also 2014 visits that have some Tier 1 and Tier 2 types present for individual Channel Units, but not all Channel Units.  E.g. Visit 2539 has 17 channel units, and Units 1-16 have Tier1/Tier2 listed, but Unit 17 doesn't. 2552 is another example--3 Units with Tier 1 and Tier 2, but the last unit (#5) doesn't.

**Comment by philipbaileynar** [comment]
See comment on #100 and re-test after full metric re-run.

**Comment by volkcj** [comment]
not resolved.

**Comment by MattReimer** [comment]
Maybe I'm not seeing it correctly. but I'm looking at the `VISIT_2552` postman json and it looks ok to me

``` json
[
    {
        "metricSchemaUrl": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics",
        "itemUrl": "https://api.CHaMPMonitoring.org/api/v1/visits/2552",
        "values": [
            {
                "name": "GenerationDate",
                "value": "2017-10-17T17:39:47.132997",
                "type": "String"
            },
            {
                "name": "ModelVersion",
                "value": "0.0.3",
                "type": "String"
            },
            {
                "name": "ChUnitNumber",
                "value": "1",
                "type": "Numeric"
            },
            {
                "name": "LWCount",
                "value": "8",
                "type": "Numeric"
            },
            {
                "name": "Tier1",
                "value": "Slow/Pool",
                "type": "String"
            },
            {
                "name": "Tier2",
                "value": "Scour Pool",
                "type": "String"
            }
        ],
        "name": "c9b429e6-e755-4dae-b769-403b7e0622b8",
        "url": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics/metrics/c9b429e6-e755-4dae-b769-403b7e0622b8"
    },
    {
        "metricSchemaUrl": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics",
        "itemUrl": "https://api.CHaMPMonitoring.org/api/v1/visits/2552",
        "values": [
            {
                "name": "GenerationDate",
                "value": "2017-10-17T17:39:47.132997",
                "type": "String"
            },
            {
                "name": "ModelVersion",
                "value": "0.0.3",
                "type": "String"
            },
            {
                "name": "ChUnitNumber",
                "value": "2",
                "type": "Numeric"
            },
            {
                "name": "LWCount",
                "value": "6",
                "type": "Numeric"
            },
            {
                "name": "Tier1",
                "value": "Small Side Channel",
                "type": "String"
            },
            {
                "name": "Tier2",
                "value": "Small Side Channel",
                "type": "String"
            }
        ],
        "name": "655ce25d-12fc-47e7-984d-6a940d45f92a",
        "url": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics/metrics/655ce25d-12fc-47e7-984d-6a940d45f92a"
    },
    {
        "metricSchemaUrl": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics",
        "itemUrl": "https://api.CHaMPMonitoring.org/api/v1/visits/2552",
        "values": [
            {
                "name": "GenerationDate",
                "value": "2017-10-17T17:39:47.132997",
                "type": "String"
            },
            {
                "name": "ModelVersion",
                "value": "0.0.3",
                "type": "String"
            },
            {
                "name": "ChUnitNumber",
                "value": "3",
                "type": "Numeric"
            },
            {
                "name": "LWCount",
                "value": "4",
                "type": "Numeric"
            },
            {
                "name": "Tier1",
                "value": "Slow/Pool",
                "type": "String"
            },
            {
                "name": "Tier2",
                "value": "Scour Pool",
                "type": "String"
            }
        ],
        "name": "b56c91e2-8992-4683-a8f2-779c0f2d556a",
        "url": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics/metrics/b56c91e2-8992-4683-a8f2-779c0f2d556a"
    },
    {
        "metricSchemaUrl": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics",
        "itemUrl": "https://api.CHaMPMonitoring.org/api/v1/visits/2552",
        "values": [
            {
                "name": "GenerationDate",
                "value": "2017-10-17T17:39:47.132997",
                "type": "String"
            },
            {
                "name": "ModelVersion",
                "value": "0.0.3",
                "type": "String"
            },
            {
                "name": "ChUnitNumber",
                "value": "5",
                "type": "Numeric"
            },
            {
                "name": "LWCount",
                "value": "4",
                "type": "Numeric"
            },
            {
                "name": "Tier1",
                "value": "Fast-Turbulent",
                "type": "String"
            },
            {
                "name": "Tier2",
                "value": "Riffle",
                "type": "String"
            }
        ],
        "name": "1928d88b-fc05-4906-8ce3-b02e256d48b2",
        "url": "https://api.CHaMPMonitoring.org/api/v1/Visit/metricschemas/QA - Aux Channel Metrics/metrics/1928d88b-fc05-4906-8ce3-b02e256d48b2"
    }
]
```

**Comment by MattReimer** [comment]
@philipbaileynar this might be the same a #100 maybe a workbench issue since the API seems fine

**Comment by philipbaileynar** [comment]
See #100 again. Tested visit 3225 for Aux and the types appear to be downloaded correctly.

**Comment by volkcj** [comment]
Can you please confirm you checked the Aux Channel Metrics schema?    I haven't been purging all metrics from schemas before reviewing. Is it possible this is a workbench data update issue?

**Comment by philipbaileynar** [comment]
I checked aux channel metric schema.

![2017-10-30_095820](https://user-images.githubusercontent.com/8645379/32184241-e90170d2-bd58-11e7-95e8-89b469fd0aab.png)

**Comment by volkcj** [comment]
see video
[Video_2017-10-30_100443.zip](https://github.com/SouthForkResearch/CHaMP_Metrics/files/1428107/Video_2017-10-30_100443.zip)

**Comment by volkcj** [comment]
I haven't been purging my database prior to downloading metrics.  I also have RBT and older metric instances in the database.  If metric purging needs to occur in the workbench prior to metric download then let me know the best functionality from the workbench to do this so that crews can do it.

**Comment by philipbaileynar** [comment]
I cannot reproduce this. Please zip and send the database that has this problem.

**Comment by volkcj** [comment]
[Workbench_CV_20171030.zip](https://github.com/SouthForkResearch/CHaMP_Metrics/files/1428239/Workbench_CV_20171030.zip)

**Comment by philipbaileynar** [comment]
1. The metrics are downloading OK.
1. The problem is that you don't have any channel unit definitions for this visit in your database.
1. The solution is to sync your CHaMP data, then the tier 1 and 2 names will appear:
    1. Open Workbench and check which DB you're connected to.
    1. Filter visit list to 3225
    1. `Data` menu and `Synchronize CHaMP Data`
    1. Opem Metric Review screen.

I followed these steps with the database provided and  this is what I saw after:

![2017-10-30_113426](https://user-images.githubusercontent.com/8645379/32188953-7b3380be-bd66-11e7-96fe-1a1c82d79c04.png)
